#!/usr/bin/env python
PACKAGE = "mav_tracker"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("sample_rectangle_size", double_t, 0, "Sample rectangle size [m]", 0.06, 0.01, 1.0)
gen.add("canny_lower_threshold", int_t, 0, "Canny lower threshold", 40, 1, 1000)
gen.add("hough_lines_accum_threshold", int_t, 0, "Hough lines P accum. threshold", 45, 0, 1000)
gen.add("hough_lines_p_accum_threshold", int_t, 0, "Hough lines P accum. threshold", 30, 0, 1000)
gen.add("hough_lines_p_min_length", int_t, 0, "Hough lines P min length [pix]", 20, 0, 200)
gen.add("hough_lines_p_max_gap", int_t, 0, "Hough lines P max gap [pix]", 10, 0, 200)
gen.add("hough_lines_distance_dial_center_threshold", double_t, 0, "Hough line filter threshold", 20.0, 0.0, 1000.0)

#gen.add("gaussian_blur_kernel_size", int_t, 0, "Gaussian blur Kernel size", 3, 1, 100)

exit(gen.generate(PACKAGE, "mav_tracker", "MavTracker"))
